modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/util/IgniteMath.java

```diff
@@ -252,9 +252,11 @@ public static long convertToLongExact(Number x) {
 
     /** Cast value to {@code long}, throwing an exception if the result overflows an {@code long}. */
     public static long convertToLongExact(double x) {
-        checkNumberLongBounds(BIGINT, x = round(x));
+        Number res = round(convertToBigDecimal(x));
```

Comment:
Why do we need additional convertation to big decimal now? I think we can do this convertation using double